### PR TITLE
chore(site): migrate a few services to react-query used in the DashboardProvider

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

File: site/src/api/queries/appearance.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const rawContent = appearance.getAttribute("content");
+    try {
+      return JSON.parse(rawContent as string);
+    } catch (ex) {
+      // Ignore this and fetch as normal!
+    }
+  }
+
+  return API.getAppearance();
+};
```
Comment (aslilac) [inline site/src/api/queries/appearance.ts:33]: I feel like this function could get confused for a query easily. can we just inline it into `appearance` or stow it somewhere else?

File: site/src/api/queries/buildInfo.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const rawContent = buildInfo.getAttribute("content");
+    try {
+      return JSON.parse(rawContent as string);
+    } catch (e) {
+      console.warn("Failed to parse build info from document", e);
+    }
+  }
+
+  return API.getBuildInfo();
+};
```
Comment (aslilac) [inline site/src/api/queries/buildInfo.ts:23]: same thing with this one

File: site/src/api/queries/entitlements.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const rawContent = entitlements.getAttribute("content");
+    try {
+      return JSON.parse(rawContent as string);
+    } catch (e) {
+      console.warn("Failed to parse entitlements from document", e);
+    }
+  }
+
+  return API.getEntitlements();
+};
```
Comment (aslilac) [inline site/src/api/queries/entitlements.ts:37]: and here

File: site/src/api/queries/experiments.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    const rawContent = experiments.getAttribute("content");
+    try {
+      return JSON.parse(rawContent as string);
+    } catch (e) {
+      console.warn("Failed to parse experiments from document", e);
+    }
+  }
+
+  return API.getExperiments();
+};
```
Comment (aslilac) [inline site/src/api/queries/experiments.ts:23]: you get the idea

**Review by aslilac — APPROVED** [review]
nice!!

File: site/src/api/queries/appearance.ts (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/api/queries/appearance.ts:33]: I created a function `getMetadataAsJSON` to abstract this a bit.

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/9598

**Pull request merged**